### ISSUE TITLE
MethodError: no method matching sumabs(::AutoGrad.Rec{Knet.KnetArray{Float64,2}}, ::Int64)

### ISSUE THREAD

**Issue by DataSmiths** [denizyuret/Knet.jl#156]
i'm running into errors using KnetArray's within a lossgradient function.  I'm running julia v0.5.0 and recently installed Knet. I get the same error from Pkg.test("Knet") -- error copied below.

Any ideas for debugging would be appreciated.

Here is the test summary:
```
Test Summary: | Pass  Error  Total
  reduction   | 1488    120   1608
ERROR: LoadError: LoadError: Some tests did not pass: 1488 passed, 0 failed, 120 errored, 0 broken.
 in finish(::Base.Test.DefaultTestSet) at ./test.jl:495
 in macro expansion; at ./test.jl:679 [inlined]
 in anonymous at ./<missing>:?
 in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading ~/julia/.julia/v0.5/Knet/test/reduction.jl, in expression starting on line 35
while loading ~/julia/.julia/v0.5/Knet/test/runtests.jl, in expression starting on line 184
==================================================================[ ERROR: Knet ]===================================================================

failed process: Process(`~/julia-3c9d75391c/bin/julia -Cx86-64 -J~/julia-3c9d75391c/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes ~/julia/.julia/v0.5/Knet/test/runtests.jl`, ProcessExited(1)) [1]

====================================================================================================================================================
INFO: Removing BaseTestNext v0.2.2
INFO: Removing Combinatorics v0.4.1
INFO: Removing IterTools v0.1.0
INFO: Removing Polynomials v0.1.6
ERROR: Knet had test errors
 in #test#61(::Bool, ::Function, ::Array{AbstractString,1}) at ./pkg/entry.jl:740
 in (::Base.Pkg.Entry.#kw##test)(::Array{Any,1}, ::Base.Pkg.Entry.#test, ::Array{AbstractString,1}) at ./<missing>:0
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{AbstractString,1}}})() at ./pkg/dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{AbstractString,1}}}, ::String) at ./file.jl:59
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at ./pkg/dir.jl:31
 in (::Base.Pkg.Dir.#kw##cd)(::Array{Any,1}, ::Base.Pkg.Dir.#cd, ::Function, ::Array{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at ./<missing>:0
 in #test#3(::Bool, ::Function, ::String, ::Vararg{String,N}) at ./pkg/pkg.jl:258
 in test(::String, ::Vararg{String,N}) at ./pkg/pkg.jl:258
```

Here is the error that is thrown by both Pkg.test("Knet") and when I try to pass KnetArrays to a lossgradient function:
```
reduction: Error During Test
  Test threw an exception of type MethodError
  Expression: gradcheck(f,gx,1; rtol=TOL1)
  MethodError: no method matching sumabs(::AutoGrad.Rec{Knet.KnetArray{Float64,2}}, ::Int64)
  Closest candidates are:
    sumabs(::AbstractArray{T,N}, ::Any) at reducedim.jl:318
    sumabs(::Knet.KnetArray{Float32,N}, ::Any) at  ~/julia/.julia/v0.5/Knet/src/reduction.jl:31
    sumabs(::Knet.KnetArray{Float64,N}, ::Any) at  ~/julia/.julia/v0.5/Knet/src/reduction.jl:31
    ...
   in (::AutoGrad.##g#21#23{Base.#sumabs,MersenneTwister})(::Array{Any,1}, ::Function, ::AutoGrad.Rec{Knet.KnetArray{Float64,2}}, ::Vararg{Any,N}) at  ~/julia/.julia/v0.5/AutoGrad/src/gradcheck.jl:140
   in (::AutoGrad.#g#22)(::AutoGrad.Rec{Knet.KnetArray{Float64,2}}, ::Vararg{Any,N}) at  ~/julia/.julia/v0.5/AutoGrad/src/gradcheck.jl:139
   in forward_pass(::Function, ::Tuple{Knet.KnetArray{Float64,2},Int64}, ::Array{Any,1}, ::Int64) at  ~/julia/.julia/v0
.5/AutoGrad/src/core.jl:88
   in (::AutoGrad.##gradfun#1#3{AutoGrad.#g#22,Int64})(::Array{Any,1}, ::Function, ::Knet.KnetArray{Float64,2}, ::Vararg{Any,N}) at ~/julia/.julia/v0.5/AutoGrad/src/core.jl:39
   in (::AutoGrad.#gradfun#2)(::Knet.KnetArray{Float64,2}, ::Vararg{Any,N}) at  ~/julia/.julia/v0.5/AutoGrad/src/core.jl:39
   in #gradcheck#17(::Array{Any,1}, ::Array{Any,1}, ::Function, ::Function, ::Knet.KnetArray{Float64,2}, ::Int64, ::Vararg{Int64,N}) at ~/julia/.julia/v0.5/AutoGrad/src/gradcheck.jl:39
   in (::AutoGrad.#kw##gradcheck)(::Array{Any,1}, ::AutoGrad.#gradcheck, ::Function, ::Knet.KnetArray{Float64,2}, ::Int64, ::Vararg{Int64,N}) at ./<missing>:0
   in macro expansion; at  ~/julia/.julia/v0.5/Knet/test/reduction.jl:47 [inlined]
   in macro expansion; at ./test.jl:672 [inlined]
   in anonymous at ./<missing>:?
   in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318


```

**Comment by denizyuret** [comment]
Are you using the last released version of Knet (0.8.3) and julia5?

**Comment by DataSmiths** [comment]
I am using Knet 0.8.3+ and juliav0.5.0
Pkg.status("Knet") returns  Knet 0.8.3+ master(dirty)

**Comment by denizyuret** [comment]
(dirty) indicates that there were some changes made to Knet code. Can you replicate the problem using a released version of Knet?  (0.8.3 with Julia5 or 0.8.4 with julia5 or julia6).

**Comment by DataSmiths** [comment]
You're right, I did make some changes to the Knet make file because I had some issues with cuda when initially getting Knet installed.  I just pulled a clean copy of Knet 0.8.4 with julia5 and got no errors with Pkg.test("Knet").  Thanks for your help!
